Cython/Compiler/Nodes.py

```diff
@@ -2742,7 +2753,8 @@ class DefNode(FuncDefNode):
     #  specialized_cpdefs   [DefNode]   list of specialized cpdef DefNodes
     #  py_cfunc_node  PyCFunctionNode/InnerFunctionNode   The PyCFunction to create and assign
     #
-    # decorator_indirection IndirectionNode Used to remove __Pyx_Method_ClassMethod for fused functions
+    # decorator_call_tree   StatList node containing LetNodes containing SimpleCallNodes
```

Comment:
I'm not so sure that the `DefNode` should know its decorator calls. They should be part of the name assignment, and not necessarily the `DefNode` itself.

Reply:
> although it's replacing something similarly intrusive.

Sorry for that. It's just that adding more tech debt on existing debt is the wrong direction. Changes in suboptimal code and designs should aim for leaving the code cleaner afterwards, even if that means more work along the way.

It's definitely not your fault – I'm totally aware that it's difficult to see what others (specifically me) might consider cruft or bad design for whatever reason, when all you're trying to do is to resolve that one problem at hand, following as good as you can whatever help and guiding principles you can extract from a fairly large code base. You're doing great work on that front, seriously!

Reply:
I found a much clearer way of doing it that I'll push in the next commit so it was a useful thing to point out anyway...